### PR TITLE
For #16351 - Make homescreen interactive when search dialog is up

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -698,6 +698,20 @@ class HomeFragment : Fragment() {
     }
 
     private fun navigateToSearch() {
+        // Dismisses the search dialog when the home content is scrolled
+        val recyclerView = sessionControlView!!.view
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:702]: Is there another way to access the `sessionControlView` without `!!`?

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt [hunk 2]
```diff
@@ -183,6 +186,15 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
 
         requireComponents.core.engine.speculativeCreateSession(isPrivate)
 
+        if (findNavController().previousBackStackEntry?.destination?.id == R.id.homeFragment) {
+            // When displayed above home, dispatches the touch events to scrim area to the HomeFragment
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:190]: 👏

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt [hunk 3]
```diff
@@ -327,6 +342,17 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         toolbarView.view.requestFocus()
     }
 
+    private fun hideDeviceKeyboard() {
+        val imm =
+            requireContext().getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
+    }
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:349]: Can we use Fragment.hideKeyboard() instead?

**Review by eliserichards — CHANGES_REQUESTED** [review]
Just a few quick questions, but looking good! I'm going to do some manual testing next.

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt (hunk 3 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:349]: No we need these flags specifically but I can add a comment to explain

**Comment by eliserichards** [comment]
## Testing

### Normal mode
✅ New tab with keyboard up -> able to click icons (top sites)
✅ New tab with keyboard up -> able to dismiss collections 

### Private mode
❌ New private tab with keyboard up -> about to click "Common myths" learn more link
image

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 1 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:702]: We bang bang everywhere we reference this on home. Not great but 🤷

**Review by eliserichards — APPROVED** [review]
Awesome job!! So impressed you got this working 😄

**Pull request merged**